app/src/main/java/com/example/henzoshimada/feeltrip/ElasticSearchController.java

```diff
@@ -64,46 +66,84 @@ protected Void doInBackground(Mood ... moods ) {
 
         @Override
         protected Void doInBackground(Mood ... moods ) {
+            if(android.os.Debug.isDebuggerConnected())
+                android.os.Debug.waitForDebugger();
             verifySettings();
 
             for (Mood mood : moods) {
                 if (!mood.isChanged()){
                     return null;
                 }
-                String moodId = mood.getId();
-                //Index index = new Index.Builder(mood).index(groupIndex).type("mood").id(moodId).build();
-                String query = "{\n";
-                Boolean changed;
+                String moodId = mood.getId(); //TODO: WHat happens if there's no valid ID? Shouldn't happen, but still
+                String query = "{\"doc\" : {";
                 // find out what fields have been changed and build query accordingly
-                for (int i = 0; i < 7; i++){
-                    changed = mood.getStateByIndex(i);
-                    if (changed){
-                        switch (i){
+                int notDone = 0;
+                for (int i = 0; i < 8; i++) { //find out how many fields were changed
+                    if (mood.getStateByIndex(i)) {
+                        notDone += 1;
+                    }
+                }
+                for (int i = 0; i < 8; i++) {
+                    if (mood.getStateByIndex(i)) {
+                        notDone -= 1;
+                        switch (i) {
                             case 0:
-                                query += ("\"mood : \"" + "\"" + mood.getMoodOption() + "\"");
+                                query += ("\"emotionalState\" : \"" + mood.getMoodOption() + "\"");
+                                if (notDone != 0) {
+                                    query += (",");
+                                }
                                 break;
                             case 1:
-                                query += ("description : " + mood.getDescription());
+                                query += ("\"description\" : \"" + mood.getDescription() + "\"");
+                                if (notDone != 0) {
+                                    query += (",");
+                                }
                                 break;
                             case 2:
-                                query += ("date : " + mood.getDate());
+                                query += ("\"date\" : " + mood.getDate().getTime());
```

Comment:
Note that here we call the .getTime() function. This is because we want to save the value in ElasticSearch as a long, not a Date. ElasticSearch has no default Date type, so saving it as a long is a very simple solution to this.